src/memray/reporters/assets/flamegraph.js

```diff
@@ -15,8 +15,42 @@ import {
 
 window.resizeMemoryGraph = resizeMemoryGraph;
 
+function packed_data_to_tree(packed_data) {
+  let strings = packed_data["strings"];
+  let nodes = packed_data["nodes"];
+  let unique_threads = packed_data["unique_threads"];
+
+  let num_nodes = nodes["name"].length;
+  let node_objects = new Array(num_nodes);
+  for (let i = 0; i < node_objects.length; ++i) {
+    node_objects[i] = {};
+  }
+
+  for (let [i, node] of node_objects.entries()) {
+    node["name"] = strings[nodes["name"][i]];
+    node["location"] = [
+      strings[nodes["function"][i]],
+      strings[nodes["filename"][i]],
+      nodes["lineno"][i],
+    ];
+    node["value"] = nodes["value"][i];
+    node["children"] = nodes["children"][i].map((idx) => node_objects[idx]);
+    node["n_allocations"] = nodes["n_allocations"][i];
+    node["thread_id"] = strings[nodes["thread_id"][i]];
+    node["interesting"] = nodes["interesting"][i] != 0;
+    node["import_system"] = nodes["import_system"][i] != 0;
+  }
+
+  let root = node_objects[0];
+  root["unique_threads"] = unique_threads.map(
+    (thread_id) => strings[thread_id]
+  );
+  return root;
+}
```

Comment:
> I think this does the same and is simpler but feel free to ignore or pick just the parts that you like:

Thanks, I learned some more JavaScript from this - I'd never seen the destructuring assignment stuff before (I'm pretty sure it didn't exist the last time I did much JS 😄).

I took most of this. I tweaked it a little bit, though. We need to maintain two copies of this function, one in Python and one in JS (since the flame graph tests need to undo our packing as well, in order to make assertions about the structure of the tree). So, I tried to make it as easy as possible to copy/paste things from JS to Python (or vice versa). I kept the same name for the local variables in both language (rather than using camel case in one and snake case in the other). I also used `obj["property"]` instead of `obj.property` for accessing object properties in the JS code, to make it easier to copy-paste into Python code where we're representing these objects as a dictionary. That makes for less idiomatic JavaScript, but easier copy-pasting.